core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala

```diff
@@ -67,6 +76,17 @@ private[ui] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
     }
   }
 
+  private def handleStopRequest(request: HttpServletRequest, action: String => Unit): Unit = {
+    if (parent.killEnabled &&
+        parent.master.securityMgr.checkModifyPermissions(request.getRemoteUser)) {
+      val name = Option(request.getParameter("name"))
```

Comment:
Don't you need to check for "terminate" parameter like done in handleKillRequest?

Reply:
Better to instead add a parameter to handleKillRequest itself (like "byName: Boolean") and change the "id" to use getParameter("name") for that case.